contracts/v0.7/bridge/optimism/OVM_EOACodeHashSet.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.6.0 <0.8.0;
+
+/* Library Imports */
+import { EnumerableSet } from "@openzeppelin/contracts/utils/EnumerableSet.sol";
+
+/* Contract Imports */
+import { ConfirmedOwner } from "@chainlink/contracts/src/v0.7/dev/ConfirmedOwner.sol";
+
+/**
+ * @dev Abstract helper contract used to keep track of OVM EOA contract set (OVM specific)
+ *
+ * The OVM implements a basic form of account abstraction. In effect, this means
+ * that the only type of account is a smart contract (no EOAs), and all user wallets
+ * are in fact smart contract wallets. So to check for EOA, we need to actually check if
+ * the sender is an OVM_ProxyEOA contract, which gets deployed by the ovmCREATEEOA opcode.
+ *
+ * As the OVM_ProxyEOA.sol contract source could potentially change in the future (i.e., due to a fork),
+ * here we actually track a set of possible EOA proxy contracts.
+ */
+abstract contract OVM_EOACodeHashSet is ConfirmedOwner {
```

Comment:
This feels like something we should pull into our repo and give you a nice library for :)

Reply:
I'd like to understand the motivation better for this.